Content.Server/Tools/ToolSystem.Welder.cs

```diff
@@ -150,7 +150,7 @@ private void OnGetMeleeDamage(EntityUid uid, WelderComponent component, ref GetM
                 return false;
 
             // Optional components.
-            Resolve(uid, ref item, ref light, ref appearance, false);
+            Resolve(uid, ref item, ref appearance, false);
```

Comment:
You should call resolvelight as if the caller doesn't pass in light it never sets it (even if possible).

Reply:
Okay done, I failed to notice that you added it in the turn welder on function